client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/SubmissionEditForm.jsx

```diff
@@ -538,44 +545,51 @@ const SubmissionEditForm = (props) => {
     />
   );
 
-  const renderSteppedQuestions = () => (
-    <Stepper
-      activeStep={stepIndex}
-      connector={<div />}
-      nonLinear
-      style={{ justifyContent: 'center', flexWrap: 'wrap', padding: 10 }}
-    >
-      {questionIds.map((id, index) => {
-        let stepButtonColor = '';
-        const isCurrentQuestion = index === stepIndex;
-        stepButtonColor = isCurrentQuestion ? blue[800] : blue[400];
-        return (
-          <Step key={id} sx={{ width: 55, height: 50 }}>
-            <StepButton
-              icon={
-                <SvgIcon htmlColor={stepButtonColor}>
-                  <circle cx="12" cy="12" r="12" />
-                  <text
-                    fill="#fff"
-                    fontSize="12"
-                    textAnchor="middle"
-                    x="12"
-                    y="16"
-                  >
-                    {index + 1}
-                  </text>
-                </SvgIcon>
-              }
-              onClick={() => {
-                setStepIndex(index);
-              }}
-              style={styles.stepButton}
-            />
-          </Step>
-        );
-      })}
-    </Stepper>
-  );
+  const renderSteppedQuestions = () => {
+    const answerId = getCurrentlySavingAnswerId();
+
+    return (
+      <Stepper
+        activeStep={stepIndex}
+        connector={<div />}
+        nonLinear
+        style={{ justifyContent: 'center', flexWrap: 'wrap', padding: 10 }}
+      >
+        {questionIds.map((id, index) => {
+          let stepButtonColor = '';
+          const isCurrentQuestion = index === stepIndex;
+          stepButtonColor = isCurrentQuestion ? blue[800] : blue[400];
+          return (
+            <Step key={id} sx={{ width: 55, height: 50 }}>
+              <StepButton
+                icon={
+                  <SvgIcon htmlColor={stepButtonColor}>
+                    <circle cx="12" cy="12" r="12" />
+                    <text
+                      fill="#fff"
+                      fontSize="12"
+                      textAnchor="middle"
+                      x="12"
+                      y="16"
+                    >
+                      {index + 1}
+                    </text>
+                  </SvgIcon>
+                }
+                onClick={() => {
+                  handleSubmit((data) =>
```

Comment:
We shouldn't need to call onSaveDraft unless the form is dirty / not yet saved.
i.e. Don't call when answer is clean, or when answer is already saved.

Reply:
understood, and done